Charts/Plot3D/Primitives/Axes/AxeBox.cs

```diff
@@ -0,0 +1,1067 @@
+namespace nzy3D.Plot3D.Primitives.Axes
+{
+
+	/// <summary>
+	/// The AxeBox displays a box with front face invisible and ticks labels.
+	/// @author Martin Pernollet
+	/// </summary>
+	public class AxeBox : IAxe
+	{
+
+		public enum AxeDirection
+		{
+			AxeX,
+			AxeY,
+			AxeZ
+		}
+
+		static internal int PRECISION = 6;
+		internal View _view;
+			// use this text renderer to get occupied volume by text
+		internal ITextRenderer _txt = new TextBitmapRenderer();
+		//Friend TextOverlay  txtRenderer;	' keep it null in order to not use it
+		//'Friend TextBillboard txt = new TextBillboard();	
+		internal IAxeLayout _layout;
+		internal BoundingBox3d _boxBounds;
+		internal BoundingBox3d _wholeBounds;
+		internal Coord3d _center;
+		internal Coord3d _scale;
+		internal float _xrange;
+		internal float _yrange;
+		internal float _zrange;
+		internal float[,] _quadx;
+		internal float[,] _quady;
+		internal float[,] _quadz;
+		internal float[] _normx;
+		internal float[] _normy;
+		internal float[] _normz;
+		internal float[,] _axeXx;
+		internal float[,] _axeXy;
+		internal float[,] _axeXz;
+		internal float[,] _axeYx;
+		internal float[,] _axeYy;
+		internal float[,] _axeYz;
+		internal float[,] _axeZx;
+		internal float[,] _axeZy;
+		internal float[,] _axeZz;
+		internal int[,] _axeXquads;
+		internal int[,] _axeYquads;
+		internal int[,] _axeZquads;
+
+		internal bool[] _quadIsHidden;
+		public AxeBox(BoundingBox3d bbox) : this(bbox, new AxeBoxLayout())
+		{
+		}
+
+		public AxeBox(BoundingBox3d bbox, IAxeLayout layout)
+		{
+			_layout = layout;
+			if (bbox.valid()) {
+				setAxe(bbox);
+			} else {
+				setAxe(new BoundingBox3d(-1, 1, -1, 1, -1, 1));
+			}
+			_wholeBounds = new BoundingBox3d();
+			Init();
+		}
+
+		public void Init()
+		{
+			this.Scale = new Coord3d(1, 1, 1);
+		}
+
+		public void Dispose()
+		{
+			//If Not IsNothing(_txtRenderer) Then
+			//  _txtRenderer.Dispose()
+			//End If
+		}
+
+		public ITextRenderer TextRenderer {
+			get { return _txt; }
+			set { _txt = value; }
+		}
+
+		public View View {
+			get { return _view; }
+			set { _view = value; }
+		}
+
+		public IAxeLayout Layout {
+			get { return _layout; }
+		}
+
+
+		public void Draw(Rendering.View.Camera camera)
+		{
+			// Set scaling
+			GL.LoadIdentity();
+			GL.Scale(_scale.x, _scale.y, _scale.z);
+			// Set culling
+			GL.Enable(EnableCap.CullFace);
+			GL.FrontFace(FrontFaceDirection.Ccw);
+			GL.CullFace(CullFaceMode.Front);
+			// Draw cube in feedback buffer for computing hidden quads
+			_quadIsHidden = getHiddenQuads(camera);
+			// Plain part of quad making the surrounding box
+			if (_layout.FaceDisplayed) {
+				Color quadcolor = _layout.QuadColor;
+				GL.PolygonMode(MaterialFace.Back, PolygonMode.Fill);
+				GL.Color4(quadcolor.r, quadcolor.g, quadcolor.b, quadcolor.a);
+				GL.LineWidth(1);
+				GL.Enable(EnableCap.PolygonOffsetFill);
+				GL.PolygonOffset(1, 1);
+				// handle stippling
+				drawCube(OpenTK.Graphics.RenderingMode.Render);
```

Comment:
`OpenTK.Graphics.RenderingMode` was renamed to `OpenTK.Graphics.OpenGL.RenderingMode`